components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -299,7 +286,7 @@ void AdsServiceImpl::Migrate() {
 void AdsServiceImpl::RegisterResourceComponentsForCurrentCountryCode() const {
   const std::string country_code = brave_l10n::GetCountryCode(local_state_);
 
-  RegisterResourceComponentsForCountryCode(country_code);
+  resource_component_->RegisterComponentForCountryCode(country_code);
```

Comment:
Should we not check if resource_component_ is null because this can happen for tests.

Reply:
Fixed